software/main.py

```diff
@@ -0,0 +1,248 @@
+                LOGI("FPS: {}, framecount: {}".format(fps, frame_cnt))
+                LOGI("ball_count: {}".format(len(processedData.balls)))
+            # End of housekeeping stuff
+
+            # Debug frame (turn off when over ssh)
+            if debug:
+                debug_frame = processedData.debug_frame
+                cv2.imshow('debug', debug_frame)
+                k = cv2.waitKey(1) & 0xff
+                if k == ord('q'):
+                    break
+            # End of housekeeping
+
+            # Main control logic uses a state machine
+            if state == "wait":
+                LOGSTATE("waiting")
+                input() # For making a break
+                state = "ball_search"
+                continue
+            # End of wait
+
+            # States used for thrower calibration
+            elif state == "input":
+                thrower_speed = int(input("Speed:")) # Input speed for thrower
+                state = "calibration"
+
+            elif state == "calibration": # Spin the motor for 10 seconds with the speed given in state "input"
+                if first_time:
+                    first_time = False
+                    start_time = time.perf_counter()
+
+                if (time.perf_counter() - start_time) > 10:
+                    print("Average:", np.average(calibration_data))
+                    print("Speed:", thrower_speed)
+                    state = "input"
+                    first_time = True
+                    calibration_data = []
+                    continue
+
+                if (processedData.basket_m.exists):
+                    print("Distance:", processedData.basket_m.distance)
+                    calibration_data.append(processedData.basket_m.distance)
+                    robot.throw(thrower_speed)
+                else: 
+                    print("No basket")
+                    continue
+            # End of states used for thrower calibration
+
+            elif state == "ball_search":
+                LOGSTATE("ball_search")
+                if len(processedData.balls) > 0:
+                    state = "ball_move"
+                    continue
+                else:
+                    robot.move(0, 0, 6)
+            # End of ball_search
+            
+            # Moving towards ball
+            elif state == "ball_move":
+                LOGSTATE("ball_move")
+                if len(processedData.balls) > 0:
+                    # Movement logic
+                    speed_x = 0
+                    speed_y = 0
+                    speed_r = 0
+
+                    # Choosing the closest ball
+                    interesting_ball = processedData.balls[-1]
+                    #print("Ball:", interesting_ball)
+
+                    if interesting_ball.x < middle_x + 25 and interesting_ball.x > middle_x - 25 and interesting_ball.distance <= 475:
+                        state = "ball_orbit"
+                        basket_ok_counter = 0
```

Comment:
You zero this counter here but there is no hint a to what it does. Maybe zero it where it is used?

Reply:
This counter is now obsolete, the whole system has been removed since it was not necessary.